extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcMetaStoreManagerFactory.java

```diff
@@ -0,0 +1,296 @@
+import org.apache.polaris.core.persistence.MetaStoreManagerFactory;
+import org.apache.polaris.core.persistence.PolarisMetaStoreManager;
+import org.apache.polaris.core.persistence.PrincipalSecretsGenerator;
+import org.apache.polaris.core.persistence.bootstrap.RootCredentialsSet;
+import org.apache.polaris.core.persistence.cache.EntityCache;
+import org.apache.polaris.core.persistence.dao.entity.BaseResult;
+import org.apache.polaris.core.persistence.dao.entity.EntityResult;
+import org.apache.polaris.core.persistence.dao.entity.PrincipalSecretsResult;
+import org.apache.polaris.core.persistence.transactional.TransactionalMetaStoreManagerImpl;
+import org.apache.polaris.core.storage.PolarisStorageIntegrationProvider;
+import org.apache.polaris.core.storage.cache.StorageCredentialCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The implementation of Configuration interface for configuring the {@link PolarisMetaStoreManager}
+ * using a JDBC backed by SQL metastore. TODO: refactor - <a
+ * href="https://github.com/apache/polaris/pull/1287/files#r2047487588">...</a>
+ */
+@ApplicationScoped
+@Identifier("relational-jdbc")
+public class JdbcMetaStoreManagerFactory implements MetaStoreManagerFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JdbcMetaStoreManagerFactory.class);
+
+  final Map<String, PolarisMetaStoreManager> metaStoreManagerMap = new HashMap<>();
+  final Map<String, StorageCredentialCache> storageCredentialCacheMap = new HashMap<>();
+  final Map<String, EntityCache> entityCacheMap = new HashMap<>();
+  final Map<String, Supplier<BasePersistence>> sessionSupplierMap = new HashMap<>();
+  protected final PolarisDiagnostics diagServices = new PolarisDefaultDiagServiceImpl();
+  // TODO: Pending discussion of if we should have one Database per realm or 1 schema per realm
+  // or realm should be a primary key on all the tables.
+  @Inject DataSource dataSource;
+  @Inject PolarisStorageIntegrationProvider storageIntegrationProvider;
+
+  protected JdbcMetaStoreManagerFactory() {}
+
+  protected PrincipalSecretsGenerator secretsGenerator(
+      RealmContext realmContext, @Nullable RootCredentialsSet rootCredentialsSet) {
+    if (rootCredentialsSet != null) {
+      return PrincipalSecretsGenerator.bootstrap(
+          realmContext.getRealmIdentifier(), rootCredentialsSet);
+    } else {
+      return PrincipalSecretsGenerator.RANDOM_SECRETS;
+    }
+  }
+
+  protected PolarisMetaStoreManager createNewMetaStoreManager() {
+    return new TransactionalMetaStoreManagerImpl();
+  }
+
+  private void initializeForRealm(
+      RealmContext realmContext, RootCredentialsSet rootCredentialsSet) {
+    DatasourceOperations databaseOperations = new DatasourceOperations(dataSource);
+    // TODO: see if we need to take script from Quarkus or can we just
+    // use the script committed in the repo.
+    try {
+      databaseOperations.executeScript("scripts/postgres/schema-v1-postgres.sql");
```

Comment:
If more that one Polaris Server run against the same database, it looks like this script will be executed by each of the servers at least ones on every startup... Is it intended to work this way?

Should schema creation happen only during bootstrap perhaps?

Reply:
Also: `schema-v1-postgres.sql` appears to be out-of-sync with current java code (e.g. it does not have realm ID columns).